### ISSUE TITLE
Does not build against PETSc 3.7.2

### ISSUE THREAD

**Issue by dpo** [dealii/dealii#2677]
Trying to upgrade the Homebrew formula for OSX, but:
```
[ 70%] Building CXX object source/lac/CMakeFiles/obj_lac.release.dir/petsc_precondition.cc.o
    ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/petscoptions.h:30:29: note: candidate function not viable: requires 3 arguments, but 2 were provided
PETSC_EXTERN PetscErrorCode PetscOptionsSetValue(PetscOptions,const char[],const char[]);
                            ^
```

**Comment by davydden** [comment]
this is being fixed in https://github.com/dealii/dealii/pull/2327, that is however currently messed up. Hopefully @onelieflet will give it another look soon.

p.s. the problem is that PETSc changed the argument list in this function between 3.6 and 3.7.

**Comment by davydden** [comment]
this is being fixed in https://github.com/dealii/dealii/pull/2327, that is however currently messed up. Hopefully @oneliefleft will give it another look soon.

p.s. the problem is that PETSc changed the argument list in this function between 3.6 and 3.7.

**Comment by oneliefleft** [comment]
Yes, this is long over due.. My bad. :-(
I will deal with it and send in the patch.

2016-06-10 7:15 GMT+02:00 Denis Davydov <notifications@github.com>:

**Comment by cpraveen** [comment]
I bumped into this issue since I upgraded petsc/slepc through homebrew. Will this be fixed soon ? How would I install older version of petsc in homebrew ?

**Comment by davydden** [comment]
@cpraveen no easy way to control version in homebrew. That's why i switched to Spack https://github.com/LLNL/spack.

**Comment by drwells** [comment]
Closed by #2762.
